subprojects/resources-http/src/test/groovy/org/gradle/internal/resource/transport/http/JavaSystemPropertiesHttpTimeoutSettingsTest.groovy

```diff
@@ -65,4 +75,15 @@ class JavaSystemPropertiesHttpTimeoutSettingsTest extends Specification {
         where:
         timeout << ["", "abc"]
     }
+
+    def "uses default value if provided idle connection timeout is not valid"() {
+        System.setProperty(IDLE_CONNECTION_TIMEOUT_SYSTEM_PROPERTY, timeout)
+        JavaSystemPropertiesHttpTimeoutSettings settings = new JavaSystemPropertiesHttpTimeoutSettings()
+
+        expect:
+        settings.idleConnectionTimeoutMs == DEFAULT_IDLE_CONNECTION_TIMEOUT
+
+        where:
+        timeout << ["", "abc"]
```

Comment:
There's no invalid number warning to check for here?

Reply:
No, we don't have one here or for any of the other values either. I think one reason for that is that these are supposed to be "internal" flags.